### PR TITLE
feat(core): Allow transferring credentials from any project to any team project

### PR DESCRIPTION
**Pull request by despairblue**
## Summary

This allows transferring credentials from any personal or team project to any other team project.

The user needs to have the permissions/scopes of `credential:move` for the credential and `credential:create` for the destination project.

Credentials cannot be moved into their current home project.
Credentials cannot be moved into any personal project.
Credentials can only be moved out of personal and team projects. (For now there are only personal and team projects, but there are plans for other kinds of projects.)

## Related tickets and issues

https://www.notion.so/n8n/Move-resources-workflow-credentials-between-projects-e737365c6ff446ce9e396df03e7a1255
https://linear.app/n8n/issue/DOC-886/feature-transfer-workflows-and-credentials-from-project-to-project
https://linear.app/n8n/issue/PAY-1428/move-resources-between-projects

## Review / Merge checklist

- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [x] Tests included.
  > A bug is not considered fixed, unless a test is added to prevent it from happening again.
  > A feature is not complete without tests.

### CODE REVIEW

File: packages/cli/src/workflows/workflow.service.ee.ts [hunk 1]
```diff
@@ -249,14 +249,14 @@ export class EnterpriseWorkflowService {
 		]);
 		NotFoundError.isDefinedAndNotNull(
 			workflow,
-			`Could not find workflow with the id "${workflowId}". Make sure you have the permission to delete it.`,
+			`Could not find workflow with the id "${workflowId}". Make sure you have the permission to move it.`,
```
Comment (despairblue) [inline packages/cli/src/workflows/workflow.service.ee.ts:252]: Unrelated fix. The error message is outdated. Initially I required the delete scope, but I created a new scope since.

File: packages/cli/test/integration/workflows/workflows.controller.ee.test.ts [hunk 2]
```diff
@@ -1340,7 +1339,7 @@ describe('PUT /:workflowId/transfer', () => {
 
 		const allSharings = await getWorkflowSharing(workflow);
 		expect(allSharings).toHaveLength(1);
-		expect(allSharings).not.toContainEqual({
+		expect(allSharings[0]).toMatchObject({
```
Comment (despairblue) [inline packages/cli/test/integration/workflows/workflows.controller.ee.test.ts:1342]: The former assertion is incorrect.

File: packages/cli/test/integration/workflows/workflows.controller.ee.test.ts
Comment (despairblue) [inline packages/cli/test/integration/workflows/workflows.controller.ee.test.ts:1]: This is mostly improving the test names and sorting the arrangements to be in the same order (source, workflow, destination)

**Review by krynble — APPROVED** [review]
LGTM

**Comment by janober** [comment]
Got released with [`n8n@1.45.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.45.0)
